kernel/Kconfig

```diff
@@ -620,6 +620,7 @@ menu "Security Options"
 
 config STACK_CANARIES
 	bool "Compiler stack canaries"
+	select ENTROPY_GENERATOR
```

Comment:
does ENTROPY_GENERATOR have any dependencies? select ignores deps, maybe we want "imply" instead? defer to @ulfalizer's opinion

Reply:
I was going to do "depends" but searched the code base and multiple subsystems use "select" to ensure inclusion of the rand/entropy stuff. This includes net and bluetooth.

Reply:
It is possible to have it without an entropy generator, there is `sys_rand32_get()` when TIMER_RANDOM_GENERATOR is enabled, though this is not a true random number generator.

Reply:
FWIW, a lot of somewhat sketchy stuff in Zephyr seems to come from people trying to be consistent with other stuff. If `depends on` will work here, then I'd say go for it, even if a thousand other things use `select`. :)

Especially if what @ceolin says is true.

Probably already seen it, but the Kconfig tips page has a [section on `select`](https://docs.zephyrproject.org/latest/guides/kconfig/index.html#select-statements).

Reply:
At least [ENTROPY_GENERATOR](https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_ENTROPY_GENERATOR.html#cmdoption-arg-config-entropy-generator) doesn't have any dependencies.

Reply:
I try to avoid select because it has caused me too many problems. If I put this as a "depends" then you need to add the ENTROPY_GENERATOR to your application and I think that is wrong. Ultimately I think the random/entropy Kconfig needs to be reworked a bit but that is something to consider for 2.2